frontends/android/BitBoxApp/app/src/main/java/ch/shiftcrypto/bitboxapp/MainActivity.java

```diff
@@ -279,6 +283,21 @@ protected void onPause() {
     }
 
     private void handleIntent(Intent intent) {
+        if (intent.getAction().equals(ACTION_USB_PERMISSION)) {
+            // See https://developer.android.com/guide/topics/connectivity/usb/host#permission-d
+            synchronized (this) {
+                UsbDevice device = (UsbDevice) intent.getParcelableExtra(UsbManager.EXTRA_DEVICE);
+                if (intent.getBooleanExtra(UsbManager.EXTRA_PERMISSION_GRANTED, false)) {
```

Comment:
Can't understand this. The docs say:

> When users reply to the dialog, your broadcast receiver receives the intent that contains the EXTRA_PERMISSION_GRANTED extra, which is a boolean representing the answer. Check this extra for a value of **true** before connecting to the device.

Is there a mistake in their example?

Reply:
```
public boolean getBooleanExtra (String name, 
                boolean defaultValue)
```

The 2nd argument is the default value, not the value to check against, so it is all correct.